docker/Dockerfile

```diff
@@ -1,15 +1,37 @@
-FROM nvcr.io/nvidia/tritonserver:22.12-py3
+ARG CUDA_VERSION=cu12
 
-RUN rm /etc/apt/sources.list.d/cuda*.list && apt-get update && apt-get install -y --no-install-recommends \
-    rapidjson-dev libgoogle-glog-dev gdb python3.8-venv \
-    && rm -rf /var/lib/apt/lists/* && cd /opt && python3 -m venv py38
+FROM nvidia/cuda:12.4.1-devel-ubuntu22.04 AS cu12
+ENV CUDA_VERSION_SHORT=cu121
```

Comment:
The base image is cuda:12.4.1, why here CUDA_VERSION_SHORT=cu121?

Reply:
it's for installing torch and only the latest torch==2.4.0 has precompiled whl packages for cu12.4, but others have for cu121
```
python3 -m pip install --no-cache-dir torch==${TORCH_VERSION} torchvision==${TORCHVISION_VERSION} --index-url https://download.pytorch.org/whl/${CUDA_VERSION_SHORT}
```